### ISSUE TITLE
Support `paths.include` in plugin config to scope kubeapilinter

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kube-api-linter#108]
### Problem

We want to run `kubeapilinter` **only on Kubernetes-style API types**, usually under `api/`, while allowing other linters (e.g., `revive`, `staticcheck`) to run project-wide.

We tried configuring this via `.golangci.yml`:

```yaml
linters:
  settings:
    custom:
      kubeapilinter:
        path: "./bin/kube-api-linter.so"
        description: "Kube API Linter plugin"
        original-url: "sigs.k8s.io/kube-api-linter"
        settings:
          lintersConfig:
            paths:
              include:
                - ^api/
```

But it fails with:

```
json: unknown field "paths"
```

### Current Workaround

We're resorting to `make lint-kube-api` with:

```bash
golangci-lint run --enable=kubeapilinter ./api/...
```

This works but isn’t ideal for teams managing linters centrally via config.

Otherwise, we see errors like

```
internal/controller/admiral_controller.go:32:2: jsontags: embedded field  is missing json tag (kubeapilinter)
	client.Client
	^
internal/controller/admiral_controller.go:33:2: jsontags: field Scheme is missing json tag (kubeapilinter)
	Scheme *runtime.Scheme
	^
```

---

### Ask

Please support a `paths.include` (or similar) field in `lintersConfig` so that `kubeapilinter` can internally filter analysis to selected paths, without affecting the global linter scope.
